### PR TITLE
feat: support config.retainOnRedirect for redirect

### PR DESCRIPTION
**Pull request by atian25**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

- ~~`duplicate` or `redirect.duplicate`?~~ `retainOnRedirect`
- should we set `duplicate` default to true ?

### CODE REVIEW

File: lib/logger.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ class Logger extends Map {
     const target = this.redirectLoggers.get(level);
     if (target) {
       target.logger.log(level, args, meta);
-      if (!target.options || !target.options.duplicate) return;
```
[the remark this answers is not in the document]
Comment (atian25) [reply lib/logger.js]: #35

**Comment by popomore** [comment]
不过 logger 上的 duplicate 可能会有歧义，看不懂是什么功能。

**Comment by atian25** [comment]
redirectDuplicate ?

**Comment by popomore** [comment]
retainOnRedirect?

**Pull request closed without merging**